### ISSUE TITLE
signals tests failing for me with perl < 5.12

### ISSUE THREAD

**Issue by pghmcfc** [plicease/Test-Script#3]
Here's what I'm getting with 5.10.1 on RHEL6:
```
$ make test
PERL_DL_NONLAZY=1 /usr/bin/perl "-MExtUtils::Command::MM" "-e" "test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
# 
# 
# 
# HARNESS_ACTIVE=1
# HARNESS_VERSION=3.35
# LANG=C
# PERL5LIB=/builddir/build/BUILD/Test-Script-1.09/blib/lib:/builddir/build/BUILD/Test-Script-1.09/blib/arch:/builddir/build/BUILD/Test-Script-1.09/inc:/builddir/build/BUILD/Test-Script-1.09/inc
# PERL_DL_NONLAZY=1
# SHELL=/bin/bash
# 
# 
# 
# PERL5LIB path
# /builddir/build/BUILD/Test-Script-1.09/blib/lib
# /builddir/build/BUILD/Test-Script-1.09/blib/arch
# /builddir/build/BUILD/Test-Script-1.09/inc
# /builddir/build/BUILD/Test-Script-1.09/inc
# 
# 
# 
# perl                  v5.10.1
# ExtUtils::MakeMaker   6.55_02
# File::Spec            3.3
# File::Spec::Functions 3.3
# IPC::Run3             0.048
# Probe::Perl           0.03
# Test::Builder         0.96
# Test::Builder::Tester 1.20
# Test::More            0.96
You should load Test::Tester before Test::Builder (or anything that loads Test::Builder) at /usr/share/perl5/vendor_perl/Test/Tester.pm line 9.
# Test::Tester          0.109
# 
# 
# 
t/00_diag.t ................. ok
t/01_compile.t .............. ok
t/02_compiles_good.t ........ ok
t/03_compiles_bad.t ......... ok
t/04_runs_good.t ............ ok
t/05_runs_bad.t ............. ok
t/06_exception.t ............ ok
    # Looks like your test exited with 9 just after 5.

#   Failed test 'compiles'
#   at t/07_signal.t line 18.
    # Looks like your test exited with 9 just after 5.

#   Failed test 'runs'
#   at t/07_signal.t line 31.
# Looks like you failed 2 tests of 2.
t/07_signal.t ............... 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/2 subtests 
    # Looks like your test exited with 1024 just after 5.

#   Failed test 'non-zero exit'
#   at t/08_trap_exit_or_signal.t line 15.
    # Looks like your test exited with 9 just after 5.

#   Failed test 'signal'
#   at t/08_trap_exit_or_signal.t line 26.
# Looks like you failed 2 tests of 2.
t/08_trap_exit_or_signal.t .. 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/2 subtests 
t/09_capture_output.t ....... ok

Test Summary Report
-------------------
t/07_signal.t             (Wstat: 512 Tests: 2 Failed: 2)
  Failed tests:  1-2
  Non-zero exit status: 2
t/08_trap_exit_or_signal.t (Wstat: 512 Tests: 2 Failed: 2)
  Failed tests:  1-2
  Non-zero exit status: 2
Files=10, Tests=31,  0 wallclock secs ( 0.06 usr  0.01 sys +  0.59 cusr  0.09 csys =  0.75 CPU)
Result: FAIL
Failed 2/10 test programs. 4/31 subtests failed.
make: *** [test_dynamic] Error 255
```
It's as if the kill in the signal script is actually killing the test itself.

I'm seeing this failure mode with all Red Hat/Fedora Linux perls up to and including 5.10.1. Everything from 5.12.4 onwards seems OK. I don't have anything inbetween to test with.

**Comment by plicease** [comment]
Thanks for the report.  I will look into this today.

**Comment by plicease** [comment]
I am unable to reproduce.  So far have tried:

 * perlbrew'd perl 5.10.1 and 5.10.0 on Debian 7
 * system perl on CentOS 6 (perl 5.10.1)

So far there is only a failure on 5.6 on cpan testers, 5.8 and 5.10 look okay.

http://fast-matrix.cpantesters.org/?dist=Test-Script%201.09

Is there anything special about your environment that might help me reproduce this?  Is this an rpm build?  Can you run the individual tests with `prove -lv`?

**Comment by pghmcfc** [comment]
Ah, I've found it. All my fault I'm afraid. I'm building rpm packages, and since EL-6 doesn't have Test::More 0.96 that's needed for subtests, I bundled copies of Test::Builder and Test::More 0.96 so that I could run the test suite. However, the signals tests use Test::Tester, which I hadn't bundled even though it's bundled with Test::Builder and Test::More in the Test-Simple dist, so I was running the CentOS system version of Test::Tester with a more recent version of Test::Builder and Test::More. I tried installing Test-Simple from CPAN in a CentOS 6 VM and that fixed it. So I'm now bundling all the modules from Test-Simple and it all works. Sorry for the noise.
